source/fuzz/transformation_record_synonymous_constants.cpp

```diff
@@ -19,8 +19,61 @@ namespace spvtools {
 namespace fuzz {
 
 namespace {
-bool IsScalarZeroConstant(const opt::analysis::Constant* constant) {
-  return constant->AsScalarConstant() && constant->IsZero();
+
+// Returns true if the two given constants are equivalent
+// (the description of IsApplicable specifies the conditions they must satisfy
+// to be considered equivalent)
+bool AreEquivalentConstants(opt::IRContext* ir_context,
+                            const opt::analysis::Constant& constant1,
+                            const opt::analysis::Constant& constant2) {
```

Comment:
Due to limitations of the constant manager and type manager, it is safer if we use the ids of the constants here - i.e. change constant1 and constant2 to be uint32_t, named constant_id1 and constant_id2, representing the ids of the constants.

Reply:
Done